packages/sources/lotus/src/transport/balance.ts

```diff
@@ -0,0 +1,141 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { AdapterResponse, sleep } from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/balance'
+import { BigNumber } from 'ethers'
+import { calculateHttpRequestKey } from '@chainlink/external-adapter-framework/cache'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+
+export type BalanceTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: {
+      method: string
+      params: string[]
+      id: number
+      jsonrpc: string
+    }
+  }
+}
+
+type RequestParams = typeof inputParameters.validated
+
+export class TotalBalanceTransport extends SubscriptionTransport<BalanceTransportTypes> {
+  requester!: Requester
+  settings!: BalanceTransportTypes['Settings']
+  endpointName!: string
+
+  async initialize(
+    dependencies: TransportDependencies<BalanceTransportTypes>,
+    _adapterSettings: BalanceTransportTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, _adapterSettings, _endpointName, transportName)
+    this.requester = dependencies.requester
+    this.settings = _adapterSettings
+    this.endpointName = _endpointName
+  }
```

Comment:
Think we can skip on setting the requester here and also remove the _ from the property names since we're using them
```suggestion
export class TotalBalanceTransport extends SubscriptionTransport<BalanceTransportTypes> {
  settings!: BalanceTransportTypes['Settings']
  endpointName!: string

  async initialize(
    dependencies: TransportDependencies<BalanceTransportTypes>,
    adapterSettings: BalanceTransportTypes['Settings'],
    endpointName: string,
    transportName: string,
  ): Promise<void> {
    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
    this.settings = adapterSettings
    this.endpointName = endpointName
  }
```

Reply:
The requester is actually used in this one.

Reply:
Ah you're right, was thinking the transport already exposed it but was mixing it up with response cache